### ISSUE TITLE
How to install kernelspec to `$CONDA_PREFIX/share/jupyter/kernels/`?

### ISSUE THREAD

**Issue by krinsman** [JuliaLang/IJulia.jl#865]
I was going to try to make a PR against the repository to change the `installkernel` function to install the kernel spec to `$CONDA_PREFIX/share/jupyter/kernels` (presumably something like `pathjoin(Env["CONDA_PREFIX"], "share", "jupyter", "kernels")` although I don't know Julia yet) whenever a version of Jupyter installed via Conda is detected, but again since I don't actually understand/know Julia I can't deduce where in the source the `installkernel` function is defined. 

Right now `installkernel` installs the kernelspec (on an Ubuntu Docker image) to [`~/.local/share/jupyter/kernels`](https://jupyter.readthedocs.io/en/latest/projects/jupyter-directories.html#id2), which amounts to essentially a (Unix-user-)global installation, whereas seemingly an installation specific to the current Conda environment should be the default, _if_ Jupyter installed via Conda is detected (which seems to already tested for in `build.jl`).

Note that I had already set `CONDA_JL_HOME=$CONDA_PREFIX` before running `Pkg.build("Conda")`, and that before running `Pkg.add("IJulia")`.

The current default behavior is an extra issue that makes IJulia unwieldy to use with a version of Julia installed from a Conda repository (e.g. conda-forge). There are of course others.

**Comment by stevengj** [comment]
> whereas seemingly an installation specific to the current Conda environment should be the default

No, the intention is to install in a user-specific global location by default, since the kernel spec works for any jupyter installation.

Indeed, the IJulia build process does not require Jupyter at all.  This way it can be used for some other Jupyter installation, or installation of a program like nteract that doesn't use Jupyter at all.

**Comment by krinsman** [comment]
Relevant for anyone finding this in the future via Google: https://github.com/JuliaLang/IJulia.jl/pull/796

**Comment by BoPeng** [comment]
@krinsman I have created a conda-forge recipe for [`sos-julia`](https://github.com/vatlab/sos-julia), which, because there is no recipe for `IJulia`, [installs `IJulia` in the recipe](https://github.com/conda-forge/staged-recipes/pull/9545/files#diff-57014213db175cd2e432bad70a5c40f8R19). If you are interested in building a recipe for `IJulia`, you could create one by following the `sos-julia` recipe (and remove the `sos-julia` part) and then I can make `sos-julia` depends on `ijulia`. Please let me know if you are interested.

https://github.com/conda-forge/staged-recipes/pull/9545

**Comment by krinsman** [comment]
@BoPeng Sorry for the really slow reply on my part.

Yes I think this would be a great idea! I think having an IJulia recipe would be really useful for a lot of people. You are correct that this was my end goal.

For the immediate future I probably don't have the bandwidth to look into this further, so by all means if you want to submit the IJulia recipe yourself, or keep your recipe for `sos-julia` as it currently is, I would have no objections to that.

**Comment by BoPeng** [comment]
The sos-julia recipe has been merged, which simply [installs iJulia with `JUPYTER_DATA_DIR` set to `${PREFIX}/share/jupyter`](https://github.com/conda-forge/sos-julia-feedstock/blob/master/recipe/meta.yaml#L17). I am not at a position to create a recipe for iJulia because I am not familiar with Julia and do not know how iJulia are released. I will however be happy to change `sos-julia` to rely on `ijulia` if you can add an `ijulia` recipe later.

Right now the only drawback is `sos-julia` will not be updated with the update of `iJulia`.
